### ISSUE TITLE
Kafka sink issue with event stream

### ISSUE THREAD

**Issue by SleepyBrett** [kubernetes/heapster#1227]
Having some issues with the kafka sink. I see the heapster-platform6-ts topic get created and data is flowing, but the event topic isn't created. I went ahead and manually created it and restarted the heapster pod, but still no joy there. 

This error message could improve. I'm trying to work my way through the code now, but any pointers to help troubleshoot this would be awesome.

Note on the log I edited the broker names to protect the innocent.

Thanks!

```
I0725 20:04:25.304370       1 heapster.go:60] /heapster --source=kubernetes:https://kubernetes.default --sink=kafka:?timeseriestopic=heapster-platform6-ts&eventstopic=heapster-platform6-ev&brokers=kafka-1....net:9092&brokers=kafka-2.....net:9092&brokers=kafka-3.....net:9092 --sink=influxdb:http://monitoring-influxdb:8086 --metric_resolution=30s --alsologtostderr
I0725 20:04:25.304445       1 heapster.go:61] Heapster version 1.1.0-beta1
I0725 20:04:25.304706       1 configs.go:60] Using Kubernetes client with master "https://kubernetes.default" and version "v1"
I0725 20:04:25.304723       1 configs.go:61] Using kubelet port 10255
I0725 20:04:25.944485       1 influxdb.go:199] created influxdb sink with options: host:monitoring-influxdb:8086 user:root db:k8s
I0725 20:04:25.944513       1 heapster.go:87] Starting with Apache Kafka Sink
I0725 20:04:25.944520       1 heapster.go:87] Starting with InfluxDB Sink
I0725 20:04:25.944531       1 heapster.go:87] Starting with Metric Sink
I0725 20:04:26.002591       1 heapster.go:166] Starting heapster on port 8082
I0725 20:04:35.000256       1 manager.go:79] Scraping metrics start: 2016-07-25 20:04:00 +0000 UTC, end: 2016-07-25 20:04:30 +0000 UTC
E0725 20:04:35.000329       1 kubelet.go:279] Node ip-172-24-134-151.us-west-2.compute.internal is not ready
I0725 20:04:36.608604       1 manager.go:152] ScrapeMetrics: time: 1.608223146s size: 653
I0725 20:04:36.807535       1 influxdb.go:177] Created database "k8s" on influxDB server at "monitoring-influxdb:8086"
I0725 20:05:05.000186       1 manager.go:79] Scraping metrics start: 2016-07-25 20:04:30 +0000 UTC, end: 2016-07-25 20:05:00 +0000 UTC
E0725 20:05:05.000256       1 kubelet.go:279] Node ip-172-24-134-151.us-west-2.compute.internal is not ready
I0725 20:05:06.702447       1 manager.go:152] ScrapeMetrics: time: 1.702165122s size: 653
W0725 20:05:26.706734       1 manager.go:119] Failed to push data to sink: Apache Kafka Sink
I0725 20:05:35.000198       1 manager.go:79] Scraping metrics start: 2016-07-25 20:05:00 +0000 UTC, end: 2016-07-25 20:05:30 +0000 UTC
E0725 20:05:35.000278       1 kubelet.go:279] Node ip-172-24-134-151.us-west-2.compute.internal is not ready
I0725 20:05:36.505609       1 manager.go:152] ScrapeMetrics: time: 1.505310763s size: 653
W0725 20:05:56.601766       1 manager.go:119] Failed to push data to sink: Apache Kafka Sink
```

Update: don't see anything in the kafka logs.

Turned the logging level up and now I see this:

```
I0725 21:54:06.856028       1 manager.go:152] ScrapeMetrics: time: 1.805651849s size: 766
I0725 21:54:06.856040       1 manager.go:154]    scrape  bucket 0: 1
I0725 21:54:06.856044       1 manager.go:154]    scrape  bucket 1: 6
I0725 21:54:06.856047       1 manager.go:154]    scrape  bucket 2: 0
I0725 21:54:06.856050       1 manager.go:154]    scrape  bucket 3: 0
I0725 21:54:06.856053       1 manager.go:154]    scrape  bucket 4: 0
I0725 21:54:06.856056       1 manager.go:154]    scrape  bucket 5: 0
I0725 21:54:06.856059       1 manager.go:154]    scrape  bucket 6: 0
I0725 21:54:06.856062       1 manager.go:154]    scrape  bucket 7: 0
I0725 21:54:06.856066       1 manager.go:154]    scrape  bucket 8: 0
I0725 21:54:06.856069       1 manager.go:154]    scrape  bucket 9: 0
I0725 21:54:06.856072       1 manager.go:154]    scrape  bucket 10: 0
I0725 21:54:06.953970       1 manager.go:113] Pushing data to: Metric Sink
I0725 21:54:06.954001       1 manager.go:116] Data push completed: Metric Sink
I0725 21:54:06.954026       1 manager.go:113] Pushing data to: Apache Kafka Sink
I0725 21:54:06.954052       1 manager.go:113] Pushing data to: InfluxDB Sink
I0725 21:54:06.954062       1 manager.go:116] Data push completed: InfluxDB Sink
W0725 21:54:26.954162       1 manager.go:119] Failed to push data to sink: Apache Kafka Sink
```

Not super helpful.

**Comment by Kokan** [comment]
I had such an issue with logging details (heapster 1.2.0 version), as it turned out the kafka producer used for heapster has an option to provide logger, but non given. All of the kafka producer logs are thrown away. You can modify the code to provide brokerConf.Logger (in case of 1.2.0 metrics/sinks/kafka/driver.go)

**Comment by DirectXMan12** [comment]
@Kokan we should probably have that modification, if we don't already

**Comment by Kokan** [comment]
The additional logging have been fixed by #1684

**Comment by DirectXMan12** [comment]
ack, closing
